server/server/views/profile.py

```diff
@@ -1,28 +1,36 @@
-from cornice.resource import resource
+from cornice.resource import resource, view
+from passlib.hash import pbkdf2_sha256
 from pyramid.security import Allow, Everyone, Authenticated, ALL_PERMISSIONS
 
 from server.models import model_to_dict
 from server.models.user import User
 
 
 @resource(collection_path='/profile', path='/profile/{profile_id}',
-          renderer='json')
+          renderer='json', cors_origins=('http://localhost:3000',))
 class UserProfile(object):
 
     def __init__(self, request, context=None):
         self.request = request
         self.context = context
+        self.owner_id = int(request.matchdict['profile_id'])
 
     def __acl__(self):
         return [(Allow, Everyone, ALL_PERMISSIONS),
-                (Allow, self.owner, 'edit')]
+                (Allow, self.owner_id, 'edit')]
 
+    @view(permission=ALL_PERMISSIONS)
```

Comment:
It not a problem but if you allow it for everyone, you don't need to add permission at all.
I had to say this in the previous comment, sorry.